### PR TITLE
SONARCS-608 NPE if there are no locations in SARIF report

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

**Comment by henryju** [comment]
LGTM

File: src/main/java/org/sonar/plugins/csharp/sarif/SarifParser10.java [hunk 1]
```diff
@@ -66,17 +69,24 @@ private static void handleIssue(JsonObject resultObj, SarifParserCallback callba
...
-
-        callback.onIssue(ruleId, absolutePath, message, startLine);
+    if (resultObj.has("locations")) {
+      for (JsonElement locationEl : resultObj.get("locations").getAsJsonArray()) {
+        JsonObject locationObj = locationEl.getAsJsonObject();
+        if (locationObj.has("resultFile")) {
+          JsonObject resultFileObj = locationObj.get("resultFile").getAsJsonObject();
+          hasLocation = true;
+
+          if (!resultFileObj.has("uri") || !resultFileObj.has("region")) {
```
Comment (SonarTech) [inline src/main/java/org/sonar/plugins/csharp/sarif/SarifParser10.java]: MAJOR Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS134)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [SarifParser10.java#L30](https://github.com/SonarSource/sonar-csharp/blob/c0e4884b8a53833c5a344d5884806d78b05b15f0/src/main/java/org/sonar/plugins/csharp/sarif/SarifParser10.java#L30): Document this public class. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUndocumentedApi)

File: src/main/java/org/sonar/plugins/csharp/sarif/SarifParser10.java [hunk 2]
```diff
@@ -66,17 +72,24 @@ private static void handleIssue(JsonObject resultObj, SarifParserCallback callba
...
-        int startLine = region.get("startLine").getAsInt();
-
-        callback.onIssue(ruleId, absolutePath, message, startLine);
+    if (resultObj.has("locations")) {
+      for (JsonElement locationEl : resultObj.get("locations").getAsJsonArray()) {
+        JsonObject locationObj = locationEl.getAsJsonObject();
+        if (locationObj.has("resultFile")) {
+          JsonObject resultFileObj = locationObj.get("resultFile").getAsJsonObject();
+
+          if (!resultFileObj.has("uri") || !resultFileObj.has("region")) {
```
Comment (SonarTech) [inline src/main/java/org/sonar/plugins/csharp/sarif/SarifParser10.java]: MAJOR Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS134)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request merged**